src/main/java/seedu/address/model/Model.java

```diff
@@ -84,4 +98,38 @@ public interface Model {
      * @throws NullPointerException if {@code predicate} is null.
      */
     void updateFilteredPersonList(Predicate<Person> predicate);
+
+    /**
+     * Returns true if an appointment with the same identity as {@code appointment} exists in CLInic.
+     */
+    boolean hasAppointment(Appointment appointment);
```

Comment:
Maybe can write hasAnotherAppointment instead to make it clearer that it returns true if duplicate